Makefile

```diff
@@ -98,18 +94,18 @@ serve-docs: check-venv
 	@. $(VENV_ACTIVATE_FILE); cd docs && $(MAKE) serve
 
 test: check-venv
-	. $(VENV_ACTIVATE_FILE); nox -s test-3.9
-	. $(VENV_ACTIVATE_FILE); nox -s test-3.12
+	uv run --python=3.9 nox -s test-3.9
+	uv run --python=3.12 nox -s test-3.12
```

Comment:
Aren't we potentially installing Python twice here? Once by `uv`, and then by `nox`? I think I prefer the previous version with `VENV_ACTIVATE` more.

Reply:
Good point.